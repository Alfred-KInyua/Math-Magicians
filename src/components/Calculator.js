import React from 'react';
import './style.css';
import './logic/Calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.onClick = this.onClick.bind(this);
  }

  render() {
    return (
      <div>
        <div className="container">
          <table className="calculator coll">
            <tr>
              <td colSpan="4">
                {' '}
                <input className="answer" type="text" id="result" disabled />
                {' '}
              </td>

            </tr>
            <tr>

              <td>
                {' '}
                <input type="button" name="AC " value="AC" onClick="display('1')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="+/-" onClick="display('2')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="%" onClick="display('3')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="รท" id="orange" onClick="display('รท')" />
                {' '}
              </td>
            </tr>
            <tr>

              <td>
                {' '}
                <input type="button" value="7" onClick="display('1')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="8" onClick="display('2')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="9" onClick="display('3')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="*" id="orange" onClick="display('/')" />
                {' '}
              </td>
            </tr>
            <tr>
              <td>
                {' '}
                <input type="button" value="4" onClick="display('4')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="5" onClick="display('5')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="6" onClick="display('6')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="-" id="orange" onClick="display('-')" />
                {' '}
              </td>
            </tr>
            <tr>
              <td>
                {' '}
                <input type="button" value="1" onClick="display('1')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="2" onClick="display('2')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="3" onClick="display('3')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="+" id="orange" onClick="display('+')" />
                {' '}
              </td>
            </tr>
            <tr>
              <td colSpan="2">
                {' '}
                <input type="button" value="0" onClick="display('0')" />
                {' '}
              </td>
              <td>
                {' '}
                <input type="button" value="*" onClick="display('*')" />
                {' '}
              </td>

              <td>
                {' '}
                <input type="button" value="=" id="orange" onClick="calculate()" />
                {' '}
              </td>

            </tr>
          </table>
        </div>
      </div>
    );
  }
}

export default Calculator;
