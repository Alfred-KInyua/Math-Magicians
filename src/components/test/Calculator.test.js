import renderer from 'react-test-renderer'; // ES6
import Calculator from '../Calculator';

describe('Testing Calculator component', () => {
  test('Testing with snapshot', () => {
    const calc = renderer.create(<Calculator />);
    expect(calc).toMatchSnapshot();
  });
});
