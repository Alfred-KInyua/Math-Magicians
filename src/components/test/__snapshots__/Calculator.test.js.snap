// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calculator component Testing with snapshot 1`] = `
<div>
  <p
    className="strong"
  >
    <strong>
      Lets do some math!
    </strong>
  </p>
  <div
    className="container"
  >
    <table
      className="calculator"
    >
      <tr>
        <td
          colSpan="4"
        >
          <div
            className="answer"
          >
             
            0
             
          </div>
           
        </td>
      </tr>
      <tr>
        <td>
          <input
            name="AC "
            onClick={[Function]}
            type="button"
            value="AC"
          />
        </td>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="+/-"
          />
           
        </td>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="%"
          />
           
        </td>
        <td>
           
          <input
            id="orange"
            onClick={[Function]}
            type="button"
            value="รท"
          />
           
        </td>
      </tr>
      <tr>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="7"
          />
           
        </td>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="8"
          />
           
        </td>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="9"
          />
           
        </td>
        <td>
           
          <input
            id="orange"
            onClick={[Function]}
            type="button"
            value="x"
          />
           
        </td>
      </tr>
      <tr>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="4"
          />
           
        </td>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="5"
          />
           
        </td>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="6"
          />
           
        </td>
        <td>
           
          <input
            id="orange"
            onClick={[Function]}
            type="button"
            value="-"
          />
           
        </td>
      </tr>
      <tr>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="1"
          />
           
        </td>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="2"
          />
           
        </td>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="3"
          />
           
        </td>
        <td>
           
          <input
            id="orange"
            onClick={[Function]}
            type="button"
            value="+"
          />
           
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
        >
           
          <input
            onClick={[Function]}
            type="button"
            value="0"
          />
           
        </td>
        <td>
           
          <input
            onClick={[Function]}
            type="button"
            value="."
          />
           
        </td>
        <td>
           
          <input
            id="orange"
            onClick={[Function]}
            type="button"
            value="="
          />
           
        </td>
      </tr>
    </table>
  </div>
</div>
`;
